helix-view/src/document.rs

```diff
@@ -1584,25 +1580,67 @@ impl Document {
     /// Intialize/updates the differ for this document with a new base.
     pub fn set_diff_base(&mut self, diff_base: Vec<u8>) {
         if let Ok((diff_base, ..)) = from_reader(&mut diff_base.as_slice(), Some(self.encoding)) {
-            if let Some(differ) = &self.diff_handle {
-                differ.update_diff_base(diff_base);
-                return;
-            }
-            self.diff_handle = Some(DiffHandle::new(diff_base, self.text.clone()))
+            self.set_diff_base_raw(diff_base);
         } else {
             self.diff_handle = None;
         }
     }
 
-    pub fn version_control_head(&self) -> Option<Arc<Box<str>>> {
-        self.version_control_head.as_ref().map(|a| a.load_full())
+    pub fn set_diff_base_raw(&mut self, diff_base: Rope) {
```

Comment:
not really sure I love the naming here. How about just `sef_diff_base` for this function and `set_diff_base_bytes` for the other function?

Rope is the standard text data type in helix not bytes so the other function should be the one with a special neame